File: Code/simd/c/simd.c
#include <stdio.h>

double add(double i,double j){
	return i+j;
}
double addsd(double i,double j ){
	__asm__ __volatile__ (
		"addsd %xmm1,%xmm0\n\t" 
		"popq %rbp\n\t"
		"retq"
	);
}

int main(void){
	double num1;
	double num2;
	for (double i=1;i<100;i++){
		for(double j=1;j<100;j++){
				num1+=add(i,j);
				num2+=addsd(i,j);
		}
	}
	printf("%f\n",num1);
	printf("%f\n",num2);
}

File: Code/simd/go/simd.go
package main

import "fmt"

func main() {
	var num1, num2 float64
	for i := float64(1); i < 100; i++ {
		for j := float64(1); j < 100; j++ {
			num1 += add(i, j)
			num2 += addsd(i, j)
		}
	}

	fmt.Println(num1, num2)
}

func add(i, j float64) float64 { return i + j }
func addsd(i, j float64) float64
